File: 782-jewels-and-stones/jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        int result = 0;
        unordered_set<char> mp(jewels.begin(), jewels.end());
        for (auto c : stones)
            if (mp.find(c) != mp.end())
                result++;
        return result;
    }
};

File: 782-jewels-and-stones/jewels-and-stones.go
func numJewelsInStones(jewels string, stones string) int {
    countJewels := 0
    typeJewels := make(map[rune]struct{})

    for _, v := range jewels {
        typeJewels[v] = struct{}{}
    }

    for _, v := range stones {
        if _, ok := typeJewels[v]; ok {
            countJewels++
        }
    }

    return countJewels
}